Source/KISAddonConfig.cs

```diff
@@ -22,52 +23,133 @@ sealed class KISAddonConfig : MonoBehaviour {
+  {
+    public bool done;
+
+    int loadedPartCount;
+    int loadedPartIndex;
+
+    IEnumerator LoadInventories ()
+    {
+      // Kerbal parts.
+      UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
+      UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
 
-    // Kerbal parts.
-    UpdateEvaPrefab(MaleKerbalEva, nodeSettings);
-    UpdateEvaPrefab(FemaleKerbalEva, nodeSettings);
-
-    // Set inventory module for every pod with crew capacity.
-    DebugEx.Info("Loading pod inventories...");
-    foreach (AvailablePart avPart in PartLoader.LoadedPartsList) {
-      if (avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
-          || avPart.name == RdKerbalEva
-          || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1) {
-        continue;
+      // Set inventory module for every pod with crew capacity.
+      DebugEx.Info("Loading pod inventories...");
+      for (loadedPartIndex = 0; loadedPartIndex < loadedPartCount; loadedPartIndex++) {
+        AvailablePart avPart = PartLoader.LoadedPartsList[loadedPartIndex];
+        if (!(avPart.name == MaleKerbalEva || avPart.name == FemaleKerbalEva
+              || avPart.name == RdKerbalEva
+              || !avPart.partPrefab || avPart.partPrefab.CrewCapacity < 1)) {
+          DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
+          AddPodInventories (avPart.partPrefab, avPart.partPrefab.CrewCapacity);
+        }
+        yield return null;
       }
+      done = true;
+    }
+
+    public override bool IsReady ()
+    {
+      return done;
+    }
 
-      DebugEx.Fine("Found part with CrewCapacity: {0}", avPart.name);
-      for (int i = 0; i < avPart.partPrefab.CrewCapacity; i++) {
-        try {
-          var moduleInventory =
-            avPart.partPrefab.AddModule(typeof(ModuleKISInventory).Name) as ModuleKISInventory;
-          KIS_Shared.AwakePartModule(moduleInventory);
-          var baseFields = new BaseFieldList(moduleInventory);
-          baseFields.Load(nodeSettings.GetNode("EvaInventory"));
-          moduleInventory.podSeat = i;
-          moduleInventory.invType = ModuleKISInventory.InventoryType.Pod;
-          DebugEx.Fine("Pod inventory module(s) for seat {0} loaded successfully", i);
-        } catch {
-          DebugEx.Error("Pod inventory module(s) for seat {0} can't be loaded!", i);
+    public override float ProgressFraction ()
+    {
+      return (float)loadedPartIndex / (float)loadedPartCount;
+    }
+
+    public override string ProgressTitle ()
+    {
+      return "Kerbal Inventory System";
+    }
+
+    public override void StartLoad ()
+    {
+      done = false;
+      loadedPartCount = PartLoader.LoadedPartsList.Count;
+      StartCoroutine (LoadInventories ());
+    }
+  }
+
+  public void Awake() {
+    List<LoadingSystem> list = LoadingScreen.Instance.loaders;
+    if (list != null) {
+      for (int i = 0; i < list.Count; i++) {
+        if (list[i] is KISConfigLoader) {
```

Comment:
Loading the configs is a cheap operation. Just load it in KISPodInventoryLoader.

Reply:
See overall reply, but I realized I did add comments.